### ISSUE TITLE
Facebook SDK in Unity Package Manager

### ISSUE THREAD

**Issue by MartinGonzalez** [facebook/facebook-sdk-for-unity#430]
It would be nice to deliver facebook sdk through UPM. We all know the benefits of that.
Is there any plan to do that?

**Comment by vicobill** [comment]
I agree! Now the .unitypackage is not so good for Unity2019.

**Comment by MartinGonzalez** [comment]
Please, give visibility to this issue

**Comment by wagenheimer** [comment]
It should first be made compatible with Unity 2019.4 that on Current Version still is not (Cannot finish an iOS build with it). And then added to Package Manager would be a must have!

**Comment by MartinGonzalez** [comment]
@wagenheimer 
Agree!, we are also facing that too, but if it was a package already, the only thing you should do when they fix is to only update the version instead of reimporting again the unitypackage that may lead to bugs/configuration errors

**Comment by MartinGonzalez** [comment]
@wagenheimer Is there an issue with that info you shared?

**Comment by wagenheimer** [comment]
@MartinGonzalez Yes, here is it. 
https://github.com/facebook/facebook-sdk-for-unity/issues/450

**Comment by ycarofm** [comment]
Package manager, yes! It would be an amazing addition

**Comment by tozpeak** [comment]
It would be great to have facebook plugin in UPM.

**Comment by albertogr91** [comment]
I think it is essential to start deprecating .unitypackages in favor of UPM packages

**Comment by ErnSur** [comment]
I would love to have UPM support.

**Comment by Vitordex** [comment]
it would be great to have this in UPM, to reduce repo size and segregate dependencies from actual code.

**Comment by pistoleta** [comment]
+1

**Comment by KolesnikovProduction** [comment]
+1

**Comment by DavidZobrist** [comment]
+1

**Comment by realluhui** [comment]
+1

**Comment by MartinGonzalez** [comment]
It would be great to have some visibility from the team, if this is in the roadmap or something...

**Comment by HanzaRu** [comment]
Surely would be good, there are alternatives but still would be better to have a official repo https://github.com/oae/unity-package-facebook-sdk

**Comment by newatlanta19006** [comment]
Adding support for this would be greatly appreciated. All the other major plugins have migrated to Unity Package Manager. Please consider prioritizing this!

**Comment by sreenaths** [comment]
+1

**Comment by ndlam71** [comment]
+1

**Comment by VasylRomanets-MoonActive** [comment]
+1

**Comment by Thaina** [comment]
When this issue will be fixed

**Comment by Razenpok** [comment]
+1

**Comment by VasylRomanets-MoonActive** [comment]
Just for your information guys: not only Google moved their Firebase SDKs to Unity Package Manager (UPM), but also OneSignal is going to provide their SDK via UPM starting from April 26th. So more and more major 3rd-party developers move their SDKs to UPM.

**Comment by danielebanovaz** [comment]
+1

**Comment by HitCache** [comment]
Not to hijack the issue thread but for @VasylRomanets-MoonActive and others who may not know yet, Google is going backwards now for some unimaginable reason. See thread: https://github.com/firebase/quickstart-unity/issues/1028 and https://github.com/firebase/quickstart-unity/issues/1030

This is obviously not the way forward and I agree Facebook and others should work on switching to UPM asap instead of the .unitypackage insanity.

**Comment by pshtif** [comment]
@HitCache yep indeed I've got calls with both Google and Unity now and there is nothing Google can do to avoid infringing upon the new Unity guidelines. However I hope Unity will revert back, I've let them know this is not the way to go and I am sure not the only one that voiced his opinion on this issue. And indeed these new guidelines would basically block Facebook SDK as well.

**Comment by Eugen1344** [comment]
At least publish it as tarball. Would be much simpler to intergrate it in our projects

**Comment by HitCache** [comment]
Or release a git repo please like proposed as a workaround for Google here: https://github.com/firebase/quickstart-unity/issues/1052

**Comment by Eugen1344** [comment]
Okay, so for anyone interested: i have created the tarball myself. And it is working perfectly. Just extract facebook SDK unitypackage  in unity, Then collect all created folders and drop them (with all meta files) somewhere in on your drive outside of unity project (let's say you drop them in FacebookSDK folder). Then put **package.json** file in your FacebookSDK folder with text:

`{
  "name": "com.facebook",
  "version": "9.2.0",
  "displayName": "Facebook SDK",
  "keywords": [
    "Facebook",
    "SDK"
  ],
  "author": {
    "name": "Facebook",
    "url": "https://developers.facebook.com/docs/unity/downloads/"
  },
  "description": "Facebook SDK (custom created package by Eugen)",
  "unity": "2017.1",
  "dependencies": {
    "com.google.external-dependency-manager": "1.2.165"
  }
}
`

This is the package definition file.
Then, to pack it in tarball, open any unity project, any script, and call
`UnityEditor.PackageManager.Client.Pack("C:/FacebookSDK", "C:/FacebookSDK");`
This will create you tarball package that you can later import in your project by inserting the following line in **manifest.json**:

`"com.facebook": "file:../Assets/Packages/com.facebook.tgz",`

(Notice this path is relative)

**Comment by TimGameDev** [comment]
Soooo, is there a chance to have UPM version for latest version?

**Comment by wagenheimer** [comment]
Is this asset dead? Tons of Bugs, the last release is almost 1 year old! =(

**Comment by CodeMasterYi** [comment]
Soooo, is there a chance to have UPM version for latest version?
Unity Scoped Registry via npmjs or openupm or Meta ones?

**Comment by CodeMasterYi** [comment]
Two years gone!

**Comment by josealvarez97** [comment]
Adding Facebook SDK via .unitypackage and Firebase packages via tarballs is a real headache, a lot of conflicts between dependencies.

**Comment by imurashka** [comment]
Hello! Any news about this request?

**Comment by CodeMasterYi** [comment]
> Hello! Any news about this request?

@imurashka Almost 3 years.

**Comment by MartinGonzalez** [comment]
Clearly they don't think in Unity developers, perhaps we will have to find another way of doing this

**Comment by CodeMasterYi** [comment]
> Clearly they don't think in Unity developers, perhaps we will have to find another way of doing this

@MartinGonzalez 
Maybe, Elon should take charge of Meta!

**Comment by MartinGonzalez** [comment]
> > Clearly they don't think in Unity developers, perhaps we will have to find another way of doing this
> 
> @MartinGonzalez Maybe, Elon should take charge of Meta!

And replace the picture with SHIBA xD

**Comment by ErnSur** [comment]
Meta is utilizing UPM for their XR packages. [Relevant Blog Post](https://developer.oculus.com/blog/unity-package-manager-sdk-updates-v59-meta-quest/)
Can we expect a similar move for FB SDK in the near future?

**Comment by NvHulst** [comment]
From a company perspective, can we get Facebook trough UPM? Would save a lot of time when updating several games integrating the SDK. 
Thanks in advance!

**Comment by Thaina** [comment]
More than 4 years and facebook still not realize that unity has UPM they should use instead of unitypackage

**Comment by ErnSur** [comment]
Meta XR packages were updated with UPM compatibility. There's even this [blog post](https://developers.meta.com/horizon/blog/unity-package-manager-sdk-updates-v59-meta-quest/) about it.

Any plans to do the same with this SDK?

**Comment by udoTayfun** [comment]
Come on guys😀 at least tell us why you are not doing this?
